src/sd/sd_message.cpp

```
@@ -42,8 +40,8 @@ namespace someip::sd {
 
 // SdEntry serialization/deserialization
 /** @implements REQ_ARCH_001, REQ_SD_001, REQ_SD_002, REQ_SD_003, REQ_SD_004, REQ_SD_005, REQ_SD_006, REQ_SD_007, REQ_SD_010, REQ_SD_011, REQ_SD_012, REQ_SD_013, REQ_SD_014, REQ_SD_020, REQ_SD_021, REQ_SD_022, REQ_SD_023, REQ_SD_024, REQ_SD_025, REQ_SD_026, REQ_SD_030, REQ_SD_031, REQ_SD_032, REQ_SD_033, REQ_SD_034, REQ_SD_035 */
-std::vector<uint8_t> SdEntry::serialize() const {
-    std::vector<uint8_t> data;
+platform::ByteBuffer SdEntry::serialize() const {
```

Cppcheck (2.21.0)

[style] The function 'get_index1' is never used.

(unusedFunction)